torch/utils/data/dataloader.py

```diff
@@ -337,18 +347,30 @@ class DataLoader(object):
             will be smaller. (default: False)
         timeout (numeric, optional): if positive, the timeout value for collecting a batch
             from workers. Should always be non-negative. (default: 0)
+        worker_init_fn (callable, optional): If not None, this will be called on each
+            worker subprocess with the worker id as input, after seeding and before data
+            loading. (default: None)
+
+    .. note:: By default, each worker will have its PyTorch seed set to a long generated
+              by main process using its RNG. You may use `torch.initial_seed()` to access
+              this value in :attr:`worker_init_fn`, which can be used to set other seeds
+              (e.g. NumPy) before data loading.
+
+    .. warning:: If ``spawn'' start method is used, :attr:`worker_init_fn` cannot be a lambda
```

Comment:
As @apaszke mentioned above, this indeed doesn't work with lambdas. So I add this warning here.

Reply:
The problem is not only with lambdas but with non-picklable functions in general (e.g. local function declared with `def` won't work either). I'd specifically mention that they have to be picklable.